controller/app/models/application.rb

```diff
@@ -1735,7 +1735,8 @@ def process_commands(result_io, component_id=nil, gear=nil)
         if gear
           pi = PortInterface.create_port_interface(gear, component_id, *command_item[:args])
           gear.port_interfaces.push(pi)
-          pi.publish_endpoint(self)
+          public_ip = command_item[:args][0]["external_address"] || gear.get_public_ip_address
+          pi.publish_endpoint(self, public_ip)
```

Comment:
Can we add a short comment? Something like, "Note: If the NOTIFY_ENDPOINT_CREATE command was generated in the process of moving a gear (as opposed to creating a new gear), the address from the gear object may not yet be updated, so we prefer to use the address specified in the command itself if it is available."